pyuvdata/uvbeam/uvbeam.py

```diff
@@ -748,9 +931,12 @@ def _fix_auto_power(self):
             # If we only have auto pol beams the data_array should be float not complex
             self.data_array = np.abs(self.data_array)
         elif np.any(pol_screen):
-            self.data_array[:, :, pol_screen] = np.abs(
-                self.data_array[:, :, pol_screen]
-            )
+            if self.future_array_shapes:
+                self.data_array[:, pol_screen] = np.abs(self.data_array[:, pol_screen])
+            else:
+                self.data_array[:, :, pol_screen] = np.abs(
+                    self.data_array[:, :, pol_screen]
+                )
```

Comment:
again, could use `...`

Reply:
Actually, here I can't because the number of axes that come after the `pol_screen` depends on whether this is a healpix or regularly gridded az/za beam.

Reply:
Yup, just hit that in my PR.